policy-controller/src/main.rs

```diff
@@ -133,9 +139,58 @@ async fn main() -> Result<()> {
         probe_networks,
     });
 
+    // Fetch the policy-controller deployment so that we can use it as an owner
+    // reference of the Lease.
+    let api = k8s::Api::<Deployment>::namespaced(runtime.client(), &control_plane_namespace);
+    let deployment = api.get(&policy_deployment_name).await?;
```

Comment:
TIOLI: we could replace the call to `get` (and similarly `patch`) with `get_metadata`. This should just work without any other changes (at least that was the case when I tested locally).

It's a bit of a micro optimisation; we'll end up holding the deployment obj in memory until the policy controller finishes running though so maybe it's worth making? wdyt?

```
# used std::mem::size_of to check runtime sizes of objs
2023-04-26T09:23:38.733221Z  INFO linkerd_policy_controller: Size of deploy meta: 416 # get_meta
2023-04-26T09:22:59.632020Z  INFO linkerd_policy_controller: Size of deploy: 2208 # get
```